### PR TITLE
SPOI-9032 Added sample application for Inner Join

### PR DESCRIPTION
**Pull request by chaithu14**

### CODE REVIEW

**Comment by chaithu14** [comment]
@bhupeshchawda @chinmaykolhatkar Please review and merge.

File: tutorials/README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 | hdfs2kafka        | Shows how to read from HDFS and write to a Kafka topic. |
 | jdbcIngest        | Shows how to read rows from a table in an SQL database and write them to a file in HDFS |
 | jdbcToJdbc        | This application reads from an input table using JDBC, converts input to user defined POJO & then writes those POJOS to another table. |
+| join              | Shows how to use streaming join operator
```
Comment (chinmaykolhatkar) [inline tutorials/README.md]: streaming inner join operator

Comment (chinmaykolhatkar) [inline tutorials/README.md]: make the name also innerjoin. Lets be explicit in examples.

File: tutorials/join/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Sample application to show how to use the POJOInnerJoinOperator to join multiple streaming data.
```
Comment (chinmaykolhatkar) [inline tutorials/join/README.md]: multiple or 2 data streams?

File: tutorials/join/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+
+Operators in sample application are as follows:
+1) POJOGenerator which emits SalesEvent/ProductEvent POJO's
+2) POJOInnerJoinOperator which joins multiple streaming POJO's and emits POJO tuple.
```
Comment (chinmaykolhatkar) [inline tutorials/join/README.md]: multiple or 2??

**Comment by chinmaykolhatkar** [comment]
Can you add into this example showcasing join operator with static and dynamic partition?

File: tutorials/README.md (hunk 1 above)
Comment (chaithu14) [inline tutorials/README.md]: Done.

**Comment by chaithu14** [comment]
Added the join operator with static partition in application. Operator doesn't have any specific logic regarding the dynamic partition. Also, I need to check whether the Managed state supports dynamic partition or not as well as the join operator.

File: tutorials/join/README.md (hunk 2 above)
Comment (chaithu14) [inline tutorials/join/README.md]: Changed to 2 streams.

File: tutorials/join/README.md (hunk 3 above)
Comment (chaithu14) [inline tutorials/join/README.md]: Done.

File: tutorials/innerjoin/src/main/java/com/example/join/InnerJoinApplication.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    dag.addStream("ProductToJoin", productGenerator.output, join.input2);
+    dag.addStream("JoinToConsole", join.outputPort, output.input);
+
+    // Setting tuple class properties to the ports of join operator
+    dag.setInputPortAttribute(join.input1, Context.PortContext.TUPLE_CLASS, POJOGenerator.SalesEvent.class);
+    dag.setInputPortAttribute(join.input2, Context.PortContext.TUPLE_CLASS, POJOGenerator.ProductEvent.class);
+    dag.setOutputPortAttribute(join.outputPort,Context.PortContext.TUPLE_CLASS, POJOGenerator.SalesEvent.class);
+
+    //static partition
+    dag.setAttribute(join, Context.OperatorContext.PARTITIONER, new StatelessPartitioner<POJOInnerJoinOperator>(3));
```
Comment (chinmaykolhatkar) [inline tutorials/innerjoin/src/main/java/com/example/join/InnerJoinApplication.java]: Can you please move this to properties.xml instead?

**Comment by chinmaykolhatkar** [comment]
@chaithu14 One last comment I've added.
Also please rebase ur branch with latest from master. I'll merge it once done.

**Comment by chaithu14** [comment]
Re-based the branch and updated.

**Pull request merged**